src/framework/piral-core/src/utils/state.test.ts

```diff
@@ -0,0 +1,20 @@
+import { withAll, withRootExtension } from './state';
+import { RootListener } from '../RootListener';
+import { Atom } from '@dbeining/react-atom';
+import { GlobalState } from '../types';
+
+describe('State Modul', () => {
+  const state: GlobalState = Atom.of({
+    app: {},
+    components: {},
+  });
+  it('withAll should return all state with dispatchers', () => {
```

Comment:
Can we make these tests a bit more useful? There is plenty that can be tested here.